project-base/phpstan.neon

```diff
@@ -2,7 +2,40 @@ parameters:
     ignoreErrors:
         # Add ignored errors here as regular expressions, e.g.:
         # - '#PHPUnit_Framework_MockObject_MockObject(.*) given#'
-        - '#Undefined variable: \$undefined#'
+        -
+            message: '#Undefined variable: \$undefined#'
+            path: *
+        # Ignore annotations in generated code
+        -
+            message: '#(PHPDoc tag @(param|return) has invalid value (.|\n)+ expected TOKEN_IDENTIFIER at offset \d+)#'
+            path: %currentWorkingDirectory%/tests/ShopBundle/Test/Codeception/_generated/AcceptanceTesterActions.php
+        -
+            message: '#(PHPDoc tag @throws with type .+ is not subtype of Throwable)#'
+            path: %currentWorkingDirectory%/tests/ShopBundle/Test/Codeception/_generated/AcceptanceTesterActions.php
+        # In tests, we often grab services using container->get or access persistent refrences using $this->getReference
+        -
+            message: '#(Property (Shopsys|Tests)\\.+::\$.+ \(.+\) does not accept object.)#'
+            path: %currentWorkingDirectory%/tests/ShopBundle/*
+        # In data fixtures, we often access persistent references using $this->getReference
+        -
+            message: '#(Property Shopsys\\.+::\$.+ \(Shopsys\\.+\) does not accept object\.)#'
+            path: %currentWorkingDirectory%/src/Shopsys/ShopBundle/DataFixtures/*
+        # A helper methods returning an array of persistent references using $this->getReference
+        -
+            message: '#Method Shopsys\\ShopBundle\\DataFixtures\\ProductDataFixtureReferenceInjector::.+\(\) should return array<.+> but returns array<string, object>\.#'
+            path: %currentWorkingDirectory%/src/Shopsys/ShopBundle/DataFixtures/ProductDataFixtureReferenceInjector.php
+        # In tests, there are helper methods for grabbing services using container->get
+        -
+            message: '#(Method .+::.+\(\) should return .+ but returns (object))#'
```

Comment:
This seems a bit broad, as it ignores return values of all methods that should return `object` in test classes. Is that OK?

Also, `symfony/symfony` changed the return value to `object|null` in `v3.4.31` (see https://github.com/shopsys/shopsys/pull/1353). Shouldn't this be reflected in the rule?

Reply:
Maybe it's because the regex matches a subset of the string, i.e. it's not `'#^(Method .+::.+\(\) should return .+ but returns (object))$#'`